components/smalltalk/squeak5/squeak5.p5m

```diff
@@ -0,0 +1,68 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2020 David Stes
+#
+
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+# see desktop/rss-glx/rss-glx.p5m:
+<transform file -> add pkg.depend.bypass-generate libGL\.so\.1>
+<transform file -> add pkg.depend.bypass-generate vm-display-X11\.so>
+
+file mode=0555 path=usr/bin/squeak
+file path=usr/doc/squeak/COPYING.gz
+file path=usr/doc/squeak/COPYRIGHT.gz
+file path=usr/doc/squeak/LICENSE.gz
+file path=usr/doc/squeak/README.Contributing.gz
+file path=usr/doc/squeak/README.Keyboard.gz
+file path=usr/doc/squeak/README.Sound.gz
+file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/B3DAcceleratorPlugin.a
+file path=usr/lib/$(MACH64)/squeak/5.0-202004291145/B3DAcceleratorPlugin.so
```

Comment:
Please remove all .a files from squeak5.p5m.
We do not deliver static libraries.

Reply:
Good point.  I just did that from the manifest.
I am not sure why they (opensmalltalk) have them installed with the install target.

Reply:
By the way "pharo" from http://pharo.org/  is not only a different Smalltalk image but I think also a different vm (virtual machine).   As far as I know, squeak5 cannot run pharo images.  But this may be incorrect understanding from my point of view.   The pharo VM itself is built from similar sources as squeak5 so it is not much work I think to add pharo as well.  But I didn't do it so far, I may be wrong.

Reply:
From the homepage (you should probably change COMPONENT_PROJECT_URL?) opensmalltalkvm.org:
"The cross-platform virtual machine for Squeak, Pharo, Cuis, and Newspeak."
I don't know how the Pharo guys build their VM, though.
But I think Pharo and CUIS would be good additions.
Probably CUIS runs out of the box with this VM because it's a straight descendent from Squeak.

Reply:
I could have a try for compiling the Pharo VM , or CUIS or Newspeak.  My understanding is that those are smalltalk dialects. I think there also exist GNU smalltalk and many other implementations, they can all be IPS packaged - if there is interest in that.   This is because Smalltalk originated in 1972 or so , so it has a long history and a history that knew many dialects and variants.   Maybe there exist more Smalltalk variants than that there exist UNIX variants ! Probably it is not a lot of effort (I may be wrong) to compile some of those VMs.   But they can be in separate packages : for example an IPS package "pharo" for the "Pharo Virtual Machine".